pkg/prometheus/promcfg.go

```diff
@@ -2297,6 +2307,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "scrape_timeout", Value: *sc.Spec.ScrapeTimeout})
 	}
 
+	if len(sc.Spec.ScrapeProtocols) > 0 {
+		cfg = cg.WithMinimumVersion("2.48.0").AppendMapItem(cfg, "scrape_protocols", sc.Spec.ScrapeProtocols)
```

Comment:
```suggestion
		cfg = cg.WithMinimumVersion("2.49.0").AppendMapItem(cfg, "scrape_protocols", sc.Spec.ScrapeProtocols)
```

Reply:
Got it! I'm waiting for the final release version.